src/mirakuru/base.py

```diff
@@ -167,7 +167,7 @@ def __enter__(self: SimpleExecutorType) -> SimpleExecutorType:
         return self.start()
 
     def __exit__(self,
-                 exc_type: Optional[Type[BaseException]],
+                 exc_type: Type[BaseException],
```

Comment:
So we support mypy strict optional only on branches requiring newer Python? Ok.

Reply:
yeah, that's for someone stuck on the early 3.5